lib/css_parser/parser.rb

```diff
@@ -122,46 +116,111 @@ def add_block!(block, options = {})
       options[:media_types] = [options[:media_types]].flatten.collect { |mt| CssParser.sanitize_media_query(mt) }
       options[:only_media_types] = [options[:only_media_types]].flatten.collect { |mt| CssParser.sanitize_media_query(mt) }
 
-      block = cleanup_block(block, options)
-
+      # TODO: Would be nice to skip this step too
       if options[:base_uri] and @options[:absolute_paths]
         block = CssParser.convert_uris(block, options[:base_uri])
       end
 
-      # Load @imported CSS
-      if @options[:import]
-        block.scan(RE_AT_IMPORT_RULE).each do |import_rule|
-          media_types = []
-          if (media_string = import_rule[-1])
-            media_string.split(',').each do |t|
-              media_types << CssParser.sanitize_media_query(t) unless t.empty?
+      current_media_queries = [:all]
+      if options[:media_types]
+        current_media_queries = options[:media_types].flatten.collect { |mt| CssParser.sanitize_media_query(mt) }
+      end
+
+      Crass.parse(block).each do |node|
+        case node
+        in node: :style_rule
```

Comment:
can we use this instead:
```
case node[:node]
when :style_rule
```

Reply:
We can but why do you think this is better? I like the pattern matching code